dashboard/src/components/dashboard.svelte

```diff
@@ -68,10 +68,23 @@
 	async function loadDevices() {
 		devices = await getDevices($authenticationData?.did);
 		const devicesWithStatus  = await getStatuses(devices);
-		devices = devices.flatMap((device, i) => [{ ...device, status: devicesWithStatus[i].status }]);
+		devices = filterDevices(devices, devicesWithStatus);
 		searchResults = devices;
 	}
 
+	function filterDevices(devices: Device[], devicesWithStatus: Device[]): Device[]{
+		return devices.flatMap((device) => {
+			let status: string;
+			devicesWithStatus.forEach((statusDevice: Device, i: number) => {
```

Comment:
Please use `.filter` and `.map` instead of forEach + delete

Reply:
filter and map are pure functions without side effects

Reply:
I'll try to remember to use pure functions too.